projects/VRDR.Tests/DeathRecord_Should.cs

```diff
@@ -1944,47 +2051,152 @@ public void Get_BirthRecordYear()
+            Assert.Equal("1961", SetterDeathRecord.BirthRecordChildYear);
+        }
+
+        [Fact]
+        public void Get_BirthRecordChildYear()
         {
+            Assert.Equal("1961", DeathRecord1_JSON.BirthRecordChildYear);
+            Assert.Equal("1940", DeathRecord1_XML.BirthRecordChildYear);
+        }
+
+        [Fact]
+        public void Set_FetalDeathRecordYear()
+        {
+            SetterDeathRecord.FetalDeathRecordYear = "1940";
+            Assert.Equal("1940", SetterDeathRecord.FetalDeathRecordYear);
+        }
+
+        [Fact]
+        public void Get_FetalDeathRecordYear()
+        {
+            Assert.Equal("1961", DeathRecord1_JSON.FetalDeathRecordYear);
+            Assert.Equal("1940", DeathRecord1_XML.FetalDeathRecordYear);
+        }
+
+        [Fact]
+        public void Set_UsualOccupationText()
+        {
+            // Check that it doesn't overload coded part
+            var occ = new Dictionary<string, string>();
+            occ["system"] = VR.CodeSystems.OccupationCDCSOC2018;
+            occ["code"] = "13-2011";
+            occ["display"]= "Accountants and Auditors";
+    
+            SetterDeathRecord.UsualOccupationCoded = occ;
+            // Set text component
             SetterDeathRecord.UsualOccupation = "secretary";
             Assert.Equal("secretary", SetterDeathRecord.UsualOccupation);
+            Assert.Equal("13-2011", SetterDeathRecord.UsualOccupationCoded["code"]);
+            Assert.Equal(VR.CodeSystems.OccupationCDCSOC2018, SetterDeathRecord.UsualOccupationCoded["system"]);
+
         }
 
         [Fact]
-        public void Get_UsualOccupation()
+        public void Get_UsualOccupationText()
         {
             Assert.Equal("Executive secretary", DeathRecord1_JSON.UsualOccupation);
             Assert.Equal("Executive secretary", DeathRecord1_XML.UsualOccupation);
 
         }
 
         [Fact]
-        public void Set_UsualIndustry()
+        public void Set_UsualIndustryText()
         {
+            // Make sure that it doesn't mess up the coded part 
+            var ind = new Dictionary<string, string>();
+            ind["system"] = VR.CodeSystems.IndustryCDCNAICS2017;
+            ind["code"] = "54121";
+            ind["display"]= "Accounting, Tax Preparation, Bookkeeping, and Payroll Services";
+            SetterDeathRecord.UsualIndustryCoded = ind;
+
+            // Check that the text value was set properly
             SetterDeathRecord.UsualIndustry = "State agency";
             Assert.Equal("State agency", SetterDeathRecord.UsualIndustry);
+
+            // Check that it didn't mess up the coded part
+            Assert.Equal("54121", SetterDeathRecord.UsualIndustryCoded["code"]);
+            Assert.Equal("Accounting, Tax Preparation, Bookkeeping, and Payroll Services", SetterDeathRecord.UsualIndustryCoded["display"]);
+            Assert.Equal(VR.CodeSystems.IndustryCDCNAICS2017, SetterDeathRecord.UsualIndustryCoded["system"]);
```

Comment:
same as above, should there be a similar test that sets text first and then adds the dictionary to make sure text is preserved?

Reply:
There is, see line 2162.